test/cmd/newapp.sh

```diff
@@ -63,10 +63,10 @@ os::cmd::expect_success_and_text 'oc new-app ruby-helloworld-sample -o yaml' 'AD
 os::cmd::expect_success_and_text 'oc new-app ruby-helloworld-sample --param MYSQL_PASSWORD=hello -o yaml' 'hello'
 
 # verify we can create from a template when some objects in the template declare an app label
-# the app label should still be applied to the other objects in the template.
-os::cmd::expect_success_and_text 'oc new-app -f test/testdata/template-with-app-label.json -o yaml' 'app: ruby-helloworld-sample'
-# verify the existing app label on an object is overridden by new-app
-os::cmd::expect_success_and_not_text 'oc new-app -f test/testdata/template-with-app-label.json -o yaml' 'app: myapp'
+# the app label will not be applied to any objects in the template.
+os::cmd::expect_success_and_not_text 'oc new-app -f test/testdata/template-with-app-label.json -o yaml' 'app: ruby-helloworld-sample'
+# verify the existing app label on an object is not overridden by new-app
+os::cmd::expect_success_and_text 'oc new-app -f test/testdata/template-with-app-label.json -o yaml' 'app: myapp'
```

Comment:
ok i thought about this some more and i guess the scenario you're thinking of is if an object already has the app label, but it's got the right value, we can go ahead and set the app label on everything else too, because there's no danger of us either:

1) replacing a label they intended to set
2) setting a mix of app labels with different values

it still feels a little debatable to me though, because you've got a template where the user made an explicit choice to put the app label on some things and not others, and now we're overriding that choice by putting the app label on everything.  i think that's got potential to cause harm, so i think i still prefer the approach of "if the app label exists on any object in the template, don't add it anywhere".

wdyt?